src/traversals/dfs.jl

```diff
@@ -1,130 +1,20 @@
-            end
-            if w != 0
-                vcolor[w] = 1
-                push!(S, w)
-            else
-                vcolor[u] = 2
-                pop!(S)
-            end
-        end
-    end
-    return false
-end
-
+### REMOVE-2.0
+@deprecate is_cyclic LightGraphs.Traversals.is_cyclic
 # Topological sort using DFS
-"""
-    topological_sort_by_dfs(g)
-
-Return a [topological sort](https://en.wikipedia.org/wiki/Topological_sorting) of a directed
-graph `g` as a vector of vertices in topological order.
-"""
-function topological_sort_by_dfs end
-# see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
 @traitfn function topological_sort_by_dfs(g::AG::IsDirected) where {T, AG<:AbstractGraph{T}}
-    vcolor = zeros(UInt8, nv(g))
-    verts = Vector{T}()
-    for v in vertices(g)
-        vcolor[v] != 0 && continue
-        S = Vector{T}([v])
-        vcolor[v] = 1
-        while !isempty(S)
-            u = S[end]
-            w = 0
-            for n in outneighbors(g, u)
-                if vcolor[n] == 1
-                    error("The input graph contains at least one loop.") # TODO 0.7 should we use a different error?
-                elseif vcolor[n] == 0
-                    w = n
-                    break
-                end
-            end
-            if w != 0
-                vcolor[w] = 1
-                push!(S, w)
-            else
-                vcolor[u] = 2
-                push!(verts, u)
-                pop!(S)
-            end
-        end
-    end
-    return reverse(verts)
+    Base.depwarn("`topological_sort_by_dfs` is deprecated. Equivalent functionality has been moved to `LightGraphs.Traversals.topological_sort`.", :toplogical_sort_by_dfs)
+    LightGraphs.Traversals.topological_sort(g, LightGraphs.Traversals.DepthFirst())
 end
 
-"""
-    dfs_tree(g, s)
-
-Return an ordered vector of vertices representing a directed acyclic graph based on
-depth-first traversal of the graph `g` starting with source vertex `s`.
-"""
-dfs_tree(g::AbstractGraph, s::Integer; dir=:out) = tree(dfs_parents(g, s; dir=dir))
-
-"""
-    dfs_parents(g, s[; dir=:out])
-
-Perform a depth-first search of graph `g` starting from vertex `s`.
-Return a vector of parent vertices indexed by vertex. If `dir` is specified,
-use the corresponding edge direction (`:in` and `:out` are acceptable values).
-
-### Implementation Notes
-This version of DFS is iterative.
-"""
-dfs_parents(g::AbstractGraph, s::Integer; dir=:out) =
-(dir == :out) ? _dfs_parents(g, s, outneighbors) : _dfs_parents(g, s, inneighbors)
+function dfs_tree(g::AbstractGraph, s::Integer; dir=:out)
+    Base.depwarn("`dfs_tree` is deprecated. Equivalent functionality has been moved to `LightGraphs.Traversals.tree`.", :dfs_tree)
+    z = dir == :out ? LightGraphs.Traversals.tree(g, s, LightGraphs.Traversals.DepthFirst()) : LightGraphs.Traversals.tree(g, s, LightGraphs.Traversals.DepthFirst(inneighbors))
+    # fixed in 2.0
+    z
```

Comment:
```suggestion
    return z
```

Reply:
Don't spend too much time on these. They're going away in 2.0.0.